app/models/git_repository.rb

```diff
@@ -114,8 +107,22 @@ def file_changed?(sha1, sha2, file)
     executor.execute!("cd #{pwd}", "git diff --quiet --name-only #{sha1}..#{sha2} #{file}")
   end
 
+  def file_content(sha, file)
+    raise ArgumentError, "Need a sha, but #{sha} (#{sha.size}) given" unless sha.size == 40
```

Comment:
FYI, the `Build` model has a validation that the `git_sha` attribute is a valid SHA1.  Might want to extract out the logic for checking that something is a valid SHA and use that, rather than checking length here, and using a regex there.